DbgShell/ColorConsoleHost.cs

```diff
@@ -1179,6 +1179,9 @@ public int Run( CommandLineParameterParser cpp )
                         }
                         else
                         {
+                            // The CLR seems to be overly reluctant to save our startup profile to disk... so it never does.
+                            // Starting a new profile, however, implicitly causes the existing profile to be written out.
+                            System.Runtime.ProfileOptimization.StartProfile( "PromptProfileData" );
```

Comment:
>StartProfile [](start = 63, length = 12)

How does one verify that this has an effect? Just observing an improvement in wall clock time? Procmon trace of it reading/writing the file?

The [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.profileoptimization.startprofile?view=netframework-4.6.2#remarks) seems to indicate that `StartProfile` won't do anything unless `SetProfileRoot` has been called first. (It could be wrong.) If we don't call that, where does it write the file, I wonder? If you want to try calling `SetProfileRoot`, my first idea at where to keep it would be next to the log files; maybe make `LogManager._GetLogDir` internal (and rename to `LogManager.GetLogDir`).

Reply:
SetProfileRoot is already getting called in MainClass.MainWorker, set to AppData\Local\Temp\DbgProvider\ProfileOpt. But no profile file was ever appearing in that folder; adding that call made it show up :)

Reply:
> But no profile file was ever appearing in that folder

Even after exiting DbgShell?

I removed this line (the new call to `StartProfile`) and launched DbgShell (under the debugger, via F5 in Visual Studio). I observed that the ProfileOpt directory was created, but was empty. Then I typed "exit" and quit DbgShell, and after it exited, I observed that the "StartupProfileData-NormalMode" file had been created (5k).

Perhaps you are closing DbgShell differently--clicking the red 'x' maybe?--and preventing the file from being written?

That said... I guess it's fine anyway to start a second profile.

---
In reply to: [264230864](https://github.com/Microsoft/DbgShell/pull/77#discussion_r264230864) [](ancestors = 264230864)